src/capi/azext_capi/helpers/os.py

```diff
@@ -22,7 +22,8 @@ def write_to_file(filename, file_input):
     """
     Writes file_input into file
     """
-    with open(filename, "w", encoding="utf-8") as manifest_file:
+    descriptor = os.open(path=filename, flags=os.O_WRONLY | os.O_CREAT | os.O_TRUNC, mode=0o600)
```

Comment:
I added this so Helm won't print warnings about the kubeconfig file being group- and world-readable.